File: PokerProtocol/src/protocol/commands/game/PlayerTurnBeganCommand.java
package protocol.commands.game;

import java.util.StringTokenizer;

import protocol.commands.Command;
import protocol.commands.ICommand;

public class PlayerTurnBeganCommand implements ICommand
{
    private final int m_playerPos;
    public static String COMMAND_NAME = "gamePLAYER_TURN_BEGAN";
    
    public PlayerTurnBeganCommand(StringTokenizer argsToken)
    {
        m_playerPos = Integer.parseInt(argsToken.nextToken());
    }
    
    public PlayerTurnBeganCommand(int pos)
    {
        m_playerPos = pos;
    }
    
    @Override
    public String encodeCommand()
    {
        final StringBuilder sb = new StringBuilder();
        sb.append(PlayerTurnBeganCommand.COMMAND_NAME);
        sb.append(Command.DELIMITER);
        sb.append(m_playerPos);
        return sb.toString();
    }
    
    public int getPlayerPos()
    {
        return m_playerPos;
    }
}

File: C#/PokerProtocol/Commands/Game/PlayerTurnBeganCommand.cs
using System.Text;
using EricUtility;
using EricUtility.Networking.Commands;

namespace PokerProtocol.Commands.Game
{
    public class PlayerTurnBeganCommand : AbstractCommand
    {
        public static string COMMAND_NAME = "gamePLAYER_TURN_BEGAN";

        private readonly int m_PlayerPos;
        private readonly int m_LastPlayerNoSeat;
        private readonly int m_MinimumRaise;

        public int PlayerPos
        {
            get { return m_PlayerPos; }
        }
        public int LastPlayerNoSeat
        {
            get { return m_LastPlayerNoSeat; }
        }
        public int MinimumRaise
        {
            get { return m_MinimumRaise; }
        }

        public PlayerTurnBeganCommand(StringTokenizer argsToken)
        {
            m_PlayerPos = int.Parse(argsToken.NextToken());
            m_LastPlayerNoSeat = int.Parse(argsToken.NextToken());
            m_MinimumRaise = int.Parse(argsToken.NextToken());
        }

        public PlayerTurnBeganCommand(int pos, int last, int minimumRaise)
        {
            m_PlayerPos = pos;
            m_LastPlayerNoSeat = last;
            m_MinimumRaise = minimumRaise;
        }

        public override void Encode(StringBuilder sb)
        {
            Append(sb, m_PlayerPos);
            Append(sb, m_LastPlayerNoSeat);
            Append(sb, m_MinimumRaise);
        }
    }
}
